Swap "share" button for "copy link" button if `navigator.share` is supported

EthanThatOneKid opened the pull request:
If the visitor's device supports the [`Navigator.share`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/share) API, then the "copy link" button is swapped out for the "share" button since their functionalities are overlapping. The "copy link" button can be seen as a fallback for visitors who are visiting on browsers that do not happen to support sharing.

Resolves #327.

EthanThatOneKid committed:
JS swap link button if `navigator.share` is supported

EthanThatOneKid committed:
`npm run lint`

mle225 reviewed (approved):
Looks good, approved 👍

EthanThatOneKid committed:
Use summary for share text instead of plain description

EthanThatOneKid committed:
Remove buggy error messages from call to `navigator.share`

EthanThatOneKid committed:
Remove buggy error messages from call to `navigator.share`

EthanThatOneKid committed:
Fixed lint error

EthanThatOneKid committed:
Only supply `share` with event URL and title

EthanThatOneKid committed:
Only supply `share` with event URL and title

EthanThatOneKid committed:
Corrected share success message

EthanThatOneKid committed:
Just allow OS to take over UX

EthanThatOneKid committed:
Merge branch 'main' into fix/327

EthanThatOneKid committed:
Ran `npm run lint`

EthanThatOneKid committed:
Found StackOverflow response to help with share jank

EthanThatOneKid committed:
I tried adding debounce

The navigator.share API is actually pretty jank. Maybe we shouldn't merge this.
